### ISSUE TITLE
Windows standalone builds works only if Visual Studio is installed

### ISSUE THREAD

**Issue by flozz** [Nuitka/Nuitka#694]
```
$  python -m nuitka --version
0.6.7
Python: 3.8.2 (tags/v3.8.2:7b3ab59, Feb 25 2020, 23:03:10) [MSC v.1916 64 bit (AMD64)]
Executable: C:\Users\vm0\Desktop\yoga\__env__\Scripts\python.exe
OS: Windows
Arch: x86_64
```

Hello,

I am trying to build a standalone version of [YOGA](https://github.com/wanadev/yoga/) for Windows. You can find my work in progress here: https://github.com/wanadev/yoga/blob/winbuild/winbuild/ .

I am able to build the standalone version without issue using the following command:

    python -m nuitka --follow-imports --standalone --windows-dependency-tool=pefile winbuild\yogawin.py

and when I test the generated executable, it works:

```
$ ./dist/yogawin.exe -h
usage: yoga [-h] {image,model} ...
[...]
```

When I transfer the result of the build on a fresh Windows machine, I have import error:

```
$ ./yogawin.exe -v
Traceback (most recent call last):
  File "C:\yogawin.dist\yogawin.py", line 3, in <module>
  File "C:\yogawin.dist\yoga\__init__.py", line 1, in <module yoga>
  File "C:\yogawin.dist\yoga\image\__init__.py", line 4, in <module yoga.image>
  File "C:\yogawin.dist\pyguetzli\__init__.py", line 1, in <module pyguetzli>
  File "C:\yogawin.dist\pyguetzli\guetzli.py", line 6, in <module pyguetzli.guetzli>
ImportError
```

... **BUT** if install Visual Studio on the machine, then it work.

Have you ever encountered this issue? What the install of Visual Studio changes on the machine to allow my standalone built app to work?

**Comment by kayhayen** [comment]
Most probably something in `gueztli.py` has a hidden dependency, that Nuitka would need to learn about. Can you make a minimal reproducer with pyguezli only, likely a simple import, then we can look into it.

**Comment by flozz** [comment]
I am able to reproduce with the following minimal example (i write here the command I used to create it using git bash on windows).

## On the build machine:

```
$ mkdir test-694
$ cd test-694/
```
```
$  virtualenv __env__
$  . __env__/Scripts/activate
$ pip install pyguetzli nuitka
```
```
$ vim test.py
import pyguetzli
print(dir(pyguetzli))
```
```
$ python test.py
['DEFAULT_JPEG_QUALITY', '__all__', '__builtins__', '__cached__', '__compiled__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__', '_guetzli', 'guetzli', 'pil_image', 'process_jpeg_bytes', 'process_pil_image', 'process_rgb_bytes']
```
```
$ python -m nuitka --follow-imports --standalone --windows-dependency-tool=pefile  --include-module=cffi test.py
Nuitka:WARNING:Unresolved '__import__' call at 'C:\Users\vm0\Desktop\test-694\__env__\lib\site-packages\cffi\verifier.py:151' may require use of '--include-plugin-directory' or '--include-plugin-files'.
```
```
$ ./test.dist/test.exe
['DEFAULT_JPEG_QUALITY', '__all__', '__builtins__', '__cached__', '__compiled__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__', '_guetzli', 'guetzli', 'pil_image', 'process_jpeg_bytes', 'process_pil_image', 'process_rgb_bytes']
(__env__)
```
![Capture d’écran de 2020-04-30 15-45-55](https://user-images.githubusercontent.com/971438/80717920-c5e79600-8af9-11ea-823c-4d19735b20b9.png)

→ Ok everything work :)

## On the test machine

I now transfer the `test.dist` folder to a fresh Windows install.

I go to the folder and I run `test.exe` using `cmd.exe` (I not installed git bash on it as it is a fresh install):

```
C:\Users\vm1\Desktop\test.dist>test
Traceback (most recent call last):
  File "C:\Users\vm1\Desktop\test.dist\test.py", line 1, in <module>
  File "C:\Users\vm1\Desktop\test.dist\pyguetzli\__init__.py", line 1, in <module pyguetzli>
  File "C:\Users\vm1\Desktop\test.dist\pyguetzli\guetzli.py", line 6, in <module pyguetzli.guetzli>
ImportError
```

![Capture d’écran de 2020-04-30 15-48-31](https://user-images.githubusercontent.com/971438/80718477-7a81b780-8afa-11ea-9f22-674ff6112e98.png)

→ do not work

## On the test machine again

Now I download and install Visual Studio for c++ dev on the test machine

![Capture d’écran de 2020-04-30 15-55-15 (copie)](https://user-images.githubusercontent.com/971438/80720262-f7ae2c00-8afc-11ea-82f4-fd6b9b6fe200.png)

And after the installation, it work:

```
C:\Users\vm1\Desktop\test.dist>test
['DEFAULT_JPEG_QUALITY', '__all__', '__builtins__', '__cached__', '__compiled__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__', '_guetzli', 'guetzli', 'pil_image', 'process_jpeg_bytes', 'process_pil_image', 'process_rgb_bytes']
```

![Capture d’écran de 2020-04-30 16-07-49 (copie)](https://user-images.githubusercontent.com/971438/80720388-1e6c6280-8afd-11ea-82fb-0ab4c898fcad.png)

-------

I can easily test things as its just VM :)

**Comment by kayhayen** [comment]
Also, the need to include cffi manually is triggered by a hidden dependency that we would like to see isolated of course too. Never a good thing if you have to do that.

**Comment by flozz** [comment]
Hello,

I tried all the following options:

    $ python -m nuitka --follow-imports --standalone --windows-dependency-tool=depends.exe --include-module=cffi test.py
    $ python -m nuitka --follow-imports --standalone --windows-dependency-tool=pefile --include-module=cffi test.py
    $ python -m nuitka --follow-imports --standalone --windows-dependency-tool=pefile --experimental=use_pefile_recurse --include-module=cffi test.py
    $ python -m nuitka --follow-imports --standalone --windows-dependency-tool=pefile --experimental=use_pefile_recurse --experimental=use_pefile_fullrecurse --include-module=cffi test.py

always with the same result:

```
C:\Users\vm1\Desktop\test.dist.depends.exe>test
Traceback (most recent call last):
  File "C:\Users\vm1\Desktop\test.dist.depends.exe\test.py", line 1, in <module>
  File "C:\Users\vm1\Desktop\test.dist.depends.exe\pyguetzli\__init__.py", line 1, in <module pyguetzli>
  File "C:\Users\vm1\Desktop\test.dist.depends.exe\pyguetzli\guetzli.py", line 6, in <module pyguetzli.guetzli>
ImportError
```

also, pefile options (recurse and fullrecurse) does not seems to change the result:

```
$ du -s test.dist.* 
28372	test.dist.depends.exe
27712	test.dist.pefile
27712	test.dist.pefile.recurse
27712	test.dist.pefile.fullrecurse
```

And the `cffi` dependency is never discovered automatically. :)

**Comment by flozz** [comment]
I made some more test, and in fact it works when I install Microsoft Visual C++ Redist (downloadable from Microsoft's website → https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads).

It is probably a feature that libs from VC Redist are not copied by nuitka... It can be considered as "standard" system libraries I suppose (I am not a Windows dev so...).

I just saw that there is info about that in the nuitka doc but I dont know why I do not find this before (https://nuitka.net/doc/user-manual.html#windows-standalone-program-redistribuation).
